C/BFS_AdjacencyList.c

```diff
@@ -0,0 +1,177 @@
+// Implementation of Breadth First Search for Graph DS (adjacency list)
+
+#include<stdio.h>
+#include<stdlib.h>
+#include<time.h>
+
+typedef struct node{
+	int dest;
+	int weight;
+	struct node* next;
+}node; // a struture to denote vertices in a edgelist
+
+typedef struct adjList{
+	int startIndex;
+	int visited;
+	node* edgeList;
+}adjList; // a structure representing the start vertex and all of its neighbours(edges)
+
+typedef struct graph{
+	int n;
+	adjList** vertices;
+}Graph; // a structure representing whole of the graph DS(adjacency list)
+
+// Queue Implementation (Required for breadth first search)
+
+typedef struct queue{

```

Comment:
Self-implemented queue is appreciable, you can also use an array-based queue for faster performance :)